implementation/src/main/java/io/smallrye/metrics/legacyapi/LegacyMetricRegistryAdapter.java

```diff
@@ -149,6 +153,7 @@ public void unRegisterApplicationMetrics(String appName) {
 
     public LegacyMetricRegistryAdapter(String scope, MeterRegistry registry, ApplicationNameResolver appNameResolver) {
         this.appNameResolver = (appNameResolver == null) ? ApplicationNameResolver.DEFAULT : appNameResolver;
+        isAppnameResolverPresent = (this.appNameResolver == null) ? false : true;
```

Comment:
based on prior line (155), it seems like this.appNameResolver would never be null.

Reply:
`ApplicationNameResolver.DEFAULT`  is the default functional interface that returns `null`.

Reply:
hmm. calling ApplicationNameResolver.DEFAULT returns null, but is the value of ApplicationNameResolver.DEFAULT itself null?